Dependencies need to be pinned in ``pyproject.toml``

As title says.

Dependencies need to be specifically pinned for ``tests`` and ``doc`` as you are testing the CICD with those optional dependencies.
Which means that all the **mandatory** dependencies as well as all the optional dependencies (the ones already in ``tests`` for instance) need to be specifically pinned in the optional dependencies.

For instance, it would look like:

```
dependencies = [
    "ansys-api-fluent>=0.4.0",
    "ansys-platform-instancemanagement~=1.1",
     .........
]

[project.optional-dependencies]
tests = [
    "ansys-api-fluent==0.x.y",
    "ansys-platform-instancemanagement==x.y.z",
     .........
    "pytest==9.1.1",
    "pytest-cov==7.1.0",
    "pytest-mock==3.15.1",
    "pytest-xdist==3.8.0",
    "pyfakefs==6.2.0"
]```

Attaching the error message that appears to be caused by an unpinned dependency.
```
tests/conftest.py:45: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.venv/lib/python3.13/site-packages/ansys/fluent/core/__init__.py:50: in <module>
    from ansys.fluent.core.launcher.launch_options import (  # noqa: F401
.venv/lib/python3.13/site-packages/ansys/fluent/core/launcher/__init__.py:26: in <module>
    from .fluent_container import (  # noqa: F401
.venv/lib/python3.13/site-packages/ansys/fluent/core/launcher/fluent_container.py:87: in <module>
    from ansys.fluent.core.launcher.launcher_utils import ComposeConfig
.venv/lib/python3.13/site-packages/ansys/fluent/core/launcher/launcher_utils.py:39: in <module>
    from ansys.fluent.core.utils.networking import find_remoting_ip
.venv/lib/python3.13/site-packages/ansys/fluent/core/utils/networking.py:35: in <module>
    from grpc_health.v1 import health_pb2, health_pb2_grpc
.venv/lib/python3.13/site-packages/grpc_health/v1/health_pb2.py:12: in <module>
    _runtime_version.ValidateProtobufRuntimeVersion(
.venv/lib/python3.13/site-packages/google/protobuf/runtime_version.py:100: in ValidateProtobufRuntimeVersion
    _ReportVersionError(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

msg = 'Detected incompatible Protobuf Gencode/Runtime versions when loading grpc_health/v1/health.proto: gencode 7.35.0 runtime 6.33.6. Runtime version cannot be older than the linked gencode version. See Protobuf version guarantees at https://protobuf.dev/support/cross-version-runtime-guarantee.'

    def _ReportVersionError(msg):
>     raise VersionError(msg)
E     google.protobuf.runtime_version.VersionError: Detected incompatible Protobuf Gencode/Runtime versions when loading grpc_health/v1/health.proto: gencode 7.35.0 runtime 6.33.6. Runtime version cannot be older than the linked gencode version. See Protobuf version guarantees at https://protobuf.dev/support/cross-version-runtime-guarantee.

.venv/lib/python3.13/site-packages/google/protobuf/runtime_version.py:50: VersionError
```
https://github.com/ansys-internal/pyfluent-mcp/actions/runs/28803287864/job/85412596614

--- a/doc/changelog.d/5235.maintenance.md
+++ b/doc/changelog.d/5235.maintenance.md
@@ -1,0 +1,1 @@
+Pin test dependencies relevant to docs and tests.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -26,10 +26,10 @@ dependencies = [
     "defusedxml>=0.7.1",
     "deprecated>=1.2.18",
     "docker>=7.1.0",
-    "grpcio>=1.30.0",
-    "grpcio-health-checking>=1.30.0",
-    "grpcio-status>=1.26.0",
-    "grpcio-reflection>=1.30.0",
+    "grpcio>=1.30.0,<1.82.0",
+    "grpcio-health-checking>=1.30.0,<1.82.0",
+    "grpcio-status>=1.26.0,<1.82.0",
+    "grpcio-reflection>=1.30.0,<1.82.0",
     "numpy>=1.14.0,<3.0.0",
     "pandas>=1.1.0,<3.0.0",
     "pandas-stubs>=1.1.0,<3.0.0",
@@ -51,7 +51,8 @@ tests = [
     "pytest-cov==7.1.0",
     "pytest-mock==3.15.1",
     "pytest-xdist==3.8.0",
-    "pyfakefs==6.2.0"
+    "pyfakefs==6.2.0",
+    { include-group = "general-all" },
 ]
 docs = [
     "Sphinx==8.1.3",
@@ -77,7 +78,18 @@ docs = [
     "python-pptx>=0.6.23",
     "quarto-cli==1.9.38",
     "pdf2image==1.17.0",
-    "seaborn>=0.13.2"
+    "seaborn>=0.13.2",
+    { include-group = "general-all" },
+]
+general-all = [
+  "ansys-platform-instancemanagement==1.1.2",
+  "ansys-tools-common==0.5.1",
+  "docker==7.1.0",
+  "grpcio==1.81.1",
+  "grpcio-health-checking==1.81.1",
+  "grpcio-status==1.81.1",
+  "grpcio-reflection==1.81.1",
+  "numpy==2.5.0",
 ]
 ci_types = [
     "basedpyright==1.39.8",

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -52,7 +52,14 @@ tests = [
     "pytest-mock==3.15.1",
     "pytest-xdist==3.8.0",
     "pyfakefs==6.2.0",
-    { include-group = "general-all" },
+    "ansys-platform-instancemanagement==1.1.2",
+    "ansys-tools-common==0.5.1",
+    "docker==7.1.0",
+    "grpcio==1.81.1",
+    "grpcio-health-checking==1.81.1",
+    "grpcio-status==1.81.1",
+    "grpcio-reflection==1.81.1",
+    "numpy==2.5.0",
 ]
 docs = [
     "Sphinx==8.1.3",
@@ -79,17 +86,14 @@ docs = [
     "quarto-cli==1.9.38",
     "pdf2image==1.17.0",
     "seaborn>=0.13.2",
-    { include-group = "general-all" },
-]
-general-all = [
-  "ansys-platform-instancemanagement==1.1.2",
-  "ansys-tools-common==0.5.1",
-  "docker==7.1.0",
-  "grpcio==1.81.1",
-  "grpcio-health-checking==1.81.1",
-  "grpcio-status==1.81.1",
-  "grpcio-reflection==1.81.1",
-  "numpy==2.5.0",
+    "ansys-platform-instancemanagement==1.1.2",
+    "ansys-tools-common==0.5.1",
+    "docker==7.1.0",
+    "grpcio==1.81.1",
+    "grpcio-health-checking==1.81.1",
+    "grpcio-status==1.81.1",
+    "grpcio-reflection==1.81.1",
+    "numpy==2.5.0",
 ]
 ci_types = [
     "basedpyright==1.39.8",

